### ISSUE TITLE
Simplify `http2https` plugin design

### ISSUE THREAD

**Issue by septs** [fatedier/frp#1879]
original configuration:
```ini
[web]
type = http
plugin = http2https
plugin_local_addr = 127.0.0.1:8080
```

simplify before:
```ini
[web]
type = http
local_ip = 127.0.0.1:8080
plugin = sslstrip
```

1. rename `http2https` to `sslstrip` or `strip_https`
2. read `local_ip` field replace `plugin_local_addr`

**Comment by fatedier** [comment]
The name is ok, we can use it as alias.

`local_ip` is conflict with `plugin`. So we can't reuse this field. `fields` with `plugin_` prefix will be the init parameters for plugins.

**Comment by septs** [comment]
@fatedier remapping `local_ip` field to `plugin_local_addr`？

**Comment by fatedier** [comment]
It's not necessary. Plugin can only use fields with `plugin_` prefix. It's not the core of frp, but only the plugin configures.

**Comment by septs** [comment]
plugin cannot read all fields of ini-section is by design?

**Comment by septs** [comment]
@fatedier `https2http` rename to `with-tls` maybe better

**Comment by septs** [comment]
Maybe can provide an alias.

`sslstrip`, `strip_https` (external name) mapping to `http2https` (internal name)
`with-tls`, `with-ssl` (external name) mapping to `https2http` (internal name)

**Comment by fatedier** [comment]
We will keep one plugin name finally.

We can make it compatible between `sslstrip` and `http2https` first. Then mark `http2https` deprecated.

In the specified future release version, we will remove the name `http2https`.

`https2http` seems ok now, we may not modify it and gather more advice.

**Comment by septs** [comment]
OK

**Comment by septs** [comment]
@fatedier 

Can you maintain uniform semantics?

`plugin_local_addr` -> `plugin_local_ip`

This can be understood as:

`plugin_local_ip` when `local_ip` with plugin

**Comment by fatedier** [comment]
`local_ip` should be `127.0.0.1`

`plugin_local_addr` should be `127.0.0.1:80`

They are different, address includes ip and port.

**Comment by bisansan** [comment]
sdfsdfdsf


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "fatedier/frp"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年10月22日(星期四) 上午7:00
收件人:&nbsp;"fatedier/frp"<frp@noreply.github.com&gt;;
抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;;
主题:&nbsp;Re: [fatedier/frp] Simplify `http2https` plugin design (#1879)





 
Closed #1879.
